### ISSUE TITLE
DataTable: [Accessibility] Column header sort doesn't work via keyboard controls

### ISSUE THREAD

**Issue by JamesYFC** [primefaces/primereact#7052]
### Describe the bug

When you have a dataTable with a sortable column, you can tab to focus the sortable header but can't use keyboard enter key to activate the sort.

### Reproducer

https://stackblitz.com/edit/pmmfmt?file=src%2FApp.jsx

### System Information

System: MacOS 14.5
Browsers: Chrome, Firefox, Safari
npmPackages: Latest

### Steps to reproduce the behavior

1. have a sortable table column such as [this](https://primereact.org/datatable/#sort)
2. tab into sortbale column header
3. enter doesn't work but clicking works

### Expected behavior

Enter should trigger the sort if focused

**Comment by benjaminhartmann** [comment]
Changing the comparison to ... === true instead of ... === 'true' seems to do the trick in the onKeyDown handler in HeaderCell:

DomHandler.getAttribute(event.currentTarget, 'data-p-sortable-column') === true

**Comment by melloware** [comment]
This is fixed please use 10.9.1
